### ISSUE TITLE
Test if HD are encrypted and recovery question

### ISSUE THREAD

**Issue by alessandro-guagnelli** [Drive-Trust-Alliance/sedutil#72]
Hello.

First of all, thanks for all your work, I was trying to understand hwo to enable hardware encryption on my drive and sedutil was what I was looking for.

I hope that this is the right place to ask a question, since it's the first time I ask a question on github.

So, I have a dual boot setup with 2 SSD.
- /dev/sda is a Samsung 850 Pro 512 GB with Windows 10, boot is on /dev/sda2
- /dev/sdb is a Samsung 850 Evo 250 GB with Arch Linux

To make it short, I made a couple of mess and risked to lose everything, but in the end with a little of trial and error I could have managed to encrypt my drive, even thou I'm not sure that I correctly enabled encryption on the second one. So, the first question is, how can I actually check if the drives are encrypted?

Second, how does the recovery works, in case I need to move the drives on another PC, or if one of the drives break?

Thanks in advance

**Comment by r0m30** [comment]
Second question first:
You can move a locked drive to another computer and use sedutil-cli to unlock it, the drive is not tied to a specific computer.  It you unlock the drive before you run a maintenance/recovery tool on the drive it will remain unlocked until you power it off.

Now the first question:

You would use the query command and the listlockingranges command:
You check that LockingEnabled = Y on the query and for RLKEna = Y  WLKEna = Y on LR0 in the listlockingranges command.

See example below.

C:\WINDOWS\system32> **sedutil-cli.exe --query \\.\PhysicalDrive0**

\\.\PhysicalDrive0 ATA Crucial_CT250MX200SSD1                   MU02             15321042FD0C
TPer function (0x0001)
    ACKNAK = N, ASYNC = N. BufferManagement = N, comIDManagement  = N, Streaming = Y, SYNC = Y
Locking function (0x0002)
    Locked = N, **LockingEnabled = Y,** LockingSupported = Y, MBRDone = Y, MBREnabled = Y, MediaEncrypt
Geometry function (0x0003)
   ............ snip ..............

C:\WINDOWS\system32> **sedutil-cli.exe --listlockingranges {Password}  \\.\PhysicalDrive0**
- 13:43:45.000 INFO: Locking Range Configuration for \\.\PhysicalDrive0
- 13:43:45.031 **INFO: LR0 Begin 0 for 0**
- 13:43:45.031 INFO:             **RLKEna = Y  WLKEna = Y**  RLocked = N  WLocked = N
- 13:43:45.062 INFO: LR1 Begin 0 for 0
- 13:43:45.062 INFO:             RLKEna = N  WLKEna = N  RLocked = N  WLocked = N

............ snip ..............

**Comment by alessandro-guagnelli** [comment]
Thank you very much! I guess that XLKEna = Y and XLocked = N means that the drive is effectively encrypted (XLKEna) but that it is unlocked, since I entered the right password (XLocked), right?

I see that the issue is now marked as question, so since I have another one I'll continue with this issue... I'm having problem with suspension, both in Windows and Linux. I see that Sleep S3 is not supported, I guess for hardware limitations (the decryption key probably get lost when the drive is shutted down, correct me if I'm wrong)... Is Sleep S4 (hibernation) supported? Technically, the computer gets powered off, so the PBA is shown again, and the drive can get the right password again.

BTW, I have another issue. When I shut down Windows using "Power off", without pressing the "Shift" key, Windows performs a Fast Shutdown, IIRC, and the kernel is "saved". Problem is, the boot loader is not shown in this case and Windows is automatically started, thus preventing me from boot Linux. Same thing with restart, but no PBA is shown. To sum up:

- Windows power off (no shift pressed): PBA is shown, but Windows starts automatically
- Windows power off (shift key pressed): PBA is shown, boot loader shows
- Windows/Archilinux reboot: No PBA shown, boot loader shows
- Archilinuz poweroff: PBA is shown, boot loader shows.

Is this related to fast boot? Do you have any idea why this is happening or it is the first time you hear about it?

**Comment by r0m30** [comment]
Yed xLKEna is the bit that indicates locking is turned on, and xLocked indicates the current locking state.

I'm not familiar with holding the shift key when powering off, in that what MS calls hybrid sleep?

I use hibernation on both Linux and Windows and SEDs, in windows I have sleep and hybrid sleep disabled.

**Comment by alessandro-guagnelli** [comment]
No, they are two different things. Hybrid sleep is like a sleep, but is also writes the ram content on the disk, so that if everything is fine, you still have a "normal wakeup" when you wake your pc (like a normal sleep), wherein case the power supply runs off (for several reasons) it's like you hibernated your PC, so even in this case nothing is lost (is just like a sleep + backup on disk, if you want).

With the shift key pressed while pressing power off, you actually doesn't notice any sensible difference (at least in my case), but you are forcing Windows NOT to use the fast startup option. With fast startup (if enabled) a power off is a log out and disconnection of any user, than a hibernation. So, when you boot up your PC, it should be a little faster since the kernel is already loaded (on disk).

I hope I was clear enough :) Anyway, I disabled fast startup (or fast boot, I don't remember correctly) and everything seems fine now!
